explicitly provide memory format when calling to clone() at SparseTensorMath.cpp

ifedan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #28688 explicitly provide memory format when calling to clone() at ProcessGroupGloo.cpp
* #28694 explicitly provide memory format when calling to clone() at observer.py
* #28691 explicitly provide memory format when calling to clone() at spectral_norm.py
* #28702 explicitly provide memory format when calling to clone() at SparseCUDATensorMath.cu
* #28701 explicitly provide memory format when calling to clone() at SparseCUDATensor.cpp
* **#28700 explicitly provide memory format when calling to clone() at SparseTensorMath.cpp**
* #28699 explicitly provide memory format when calling to clone() at SparseTensor.cpp
* #28698 explicitly provide memory format when calling to clone() at TensorShape.cpp
* #28697 explicitly provide memory format when calling to clone() at SparseTensorUtils.h

Differential Revision: [D18333349](https://our.internmc.facebook.com/intern/diff/D18333349)

ifedan committed:
explicitly provide memory format when calling to clone() at SparseTensorMath.cpp

[ghstack-poisoned]

ifedan committed:
Update on "explicitly provide memory format when calling to clone() at SparseTensorMath.cpp"




[ghstack-poisoned]

VitalyFedyunin reviewed (approved):
Can be preserve, but doesn't matter much

ifedan committed:
Update on "explicitly provide memory format when calling to clone() at SparseTensorMath.cpp"


Differential Revision: [D18333349](https://our.internmc.facebook.com/intern/diff/D18333349)

[ghstack-poisoned]

ifedan committed:
Update on "explicitly provide memory format when calling to clone() at SparseTensorMath.cpp"


Differential Revision: [D18333349](https://our.internmc.facebook.com/intern/diff/D18333349)

[ghstack-poisoned]

ifedan committed:
Update on "explicitly provide memory format when calling to clone() at SparseTensorMath.cpp"


Differential Revision: [D18333349](https://our.internmc.facebook.com/intern/diff/D18333349)

[ghstack-poisoned]

ifedan committed:
Update on "explicitly provide memory format when calling to clone() at SparseTensorMath.cpp"


Differential Revision: [D18333349](https://our.internmc.facebook.com/intern/diff/D18333349)

[ghstack-poisoned]

ifedan committed:
Update on "explicitly provide memory format when calling to clone() at SparseTensorMath.cpp"


Differential Revision: [D18333349](https://our.internmc.facebook.com/intern/diff/D18333349)

[ghstack-poisoned]
